go.mod

```diff
@@ -3,61 +3,64 @@ module github.com/algorand/go-algorand
 go 1.17
 
 require (
+	github.com/DataDog/zstd v1.5.2
 	github.com/algorand/avm-abi v0.1.0
 	github.com/algorand/falcon v0.0.0-20220727072124-02a2a64c4414
 	github.com/algorand/go-codec/codec v1.1.8
 	github.com/algorand/go-deadlock v0.2.2
 	github.com/algorand/go-sumhash v0.1.0
 	github.com/algorand/graphtrace v0.1.0
 	github.com/algorand/msgp v1.1.52
-	github.com/algorand/oapi-codegen v1.3.7
+	github.com/algorand/oapi-codegen v1.12.0-algorand.0
 	github.com/algorand/websocket v1.4.5
 	github.com/aws/aws-sdk-go v1.16.5
 	github.com/consensys/gnark-crypto v0.7.0
 	github.com/davidlazar/go-crypto v0.0.0-20170701192655-dcfb0a7ac018
 	github.com/dchest/siphash v1.2.1
 	github.com/fatih/color v1.7.0
 	github.com/gen2brain/beeep v0.0.0-20180718162406-4e430518395f
-	github.com/getkin/kin-openapi v0.22.0
+	github.com/getkin/kin-openapi v0.107.0
 	github.com/gofrs/flock v0.7.0
 	github.com/golang/snappy v0.0.4
 	github.com/google/go-querystring v1.0.0
-	github.com/gorilla/mux v1.6.2
+	github.com/gorilla/mux v1.8.0
 	github.com/jmoiron/sqlx v1.2.0
 	github.com/karalabe/usb v0.0.2
-	github.com/labstack/echo/v4 v4.1.17
+	github.com/labstack/echo/v4 v4.9.1
 	github.com/mattn/go-sqlite3 v1.10.0
 	github.com/miekg/dns v1.1.27
 	github.com/olivere/elastic v6.2.14+incompatible
 	github.com/sirupsen/logrus v1.8.1
 	github.com/spf13/cobra v0.0.3
-	github.com/stretchr/testify v1.7.1
-	golang.org/x/crypto v0.0.0-20220321153916-2c7772ba3064
-	golang.org/x/sys v0.0.0-20220319134239-a9b59b0215f8
-	golang.org/x/text v0.3.7
+	github.com/stretchr/testify v1.8.1
+	golang.org/x/crypto v0.1.0
```

Comment:
@Eric-Warehime The PR presents a number of dependency version changes beyond _oapi-codegen_.  How should the changes be interpreted?

* I'm trying to gauge if all changes are _necessary_.  And for necessary changes, I'm gauging the level of review performed.
* I'm asking without (yet) having researched specific changes.

Reply:
These are all packages which are either direct or indirect dependencies of `oapi-codegen`. The ones in this section are also direct dependencies of `go-algorand`. 

In terms of upgrading them, it's just a result of running `go get github.com/algorand/oapi-codegen@v1.12.0-algorand.0`. I haven't personally gone through every one of them to examine the changelogs. Most of the direct dep updates relate to our API (echo, mux, kin), testify is a test dep, and the rest are standard go modules.

It seems to me that upgrading is purely to our advantage for these sorts of things. For example, here is the diff for our crypto dep which shows 5 security vulnerabilities coming from transitive dependencies https://deps.dev/go/golang.org%2Fx%2Fcrypto/v0.0.0-20220321153916-2c7772ba3064/compare?v2=v0.1.0

Let me know if you want me to look into any of these more and I'm happy to.

Reply:
@Eric-Warehime Got it - I'm open to being told I'm off-base, but my current thinking is:  
* For any dependency used in transaction processing, I think we must closely review diffs for unexpected behavior changes.
* During this pass, I didn't have time to vet dependencies.  I'll aim to do so, but I want to leave the note in case there's different opinions + so that others can independently verify.

Rationale:
* Generally, I agree with your perspective - Stale dependencies are often problematic.
* However, for transaction processing logic, I think we must confirm that APIs go-algorand already uses do _not_ have behavior changes.  
  * If we missed a behavior change, it's possible that node 1 running dependency A@v1.0.0 disagrees with node 2 running A@v1.1.0.
  * Here's a previous example that's heightened my attention to the concern:  https://github.com/algorand/go-algorand/pull/4368#issuecomment-1209808072.

Reply:
Checking the following packages seems reasonable:
* golang.org/x/crypto
* golang.org/x/sys
* golang.org/x/text
* github.com/gorilla/mux

# golang.org/x/crypto
I don't see commits that touch any of the packages we're using.
https://github.com/golang/crypto/commits/master

# golang.org/x/sys
This is for OS specific helpers. We're mostly OS agnostic, in the cases we aren't I don't see a lot of overlap in the changes to this library.
https://github.com/golang/sys/commits/master?before=fc697a31fa06b616162e34fd66047ab52722ba6c+35&branch=master

# golang.org/x/text
This is only used in test code for `data/transactions/logic/jsonspec_test.go`
There are also very few changes in the history and I don't see anything relevant to what we're using.
https://github.com/golang/text/commits/master

# github.com/gorilla/mux
This definitely has the most changes, but they seem to be mostly new features and performance improvements. Out of these 4 this is probably the biggest risk.
https://github.com/gorilla/mux/releases
https://github.com/gorilla/mux/commits/master

Reply:
@Eric-Warehime I also performed a review and found _no_ concerns for transaction processing.  If you're comfortable with the reviews performed by @winder and me, then feel welcomed to resolve the thread. 

Methodology:
* For each dependency below, I reviewed release notes and associated commit messages.  
  * https://github.com/labstack/echo/compare/v4.1.17...v4.9.1
  * https://github.com/gorilla/mux/compare/v1.6.2...v1.8.0
  * https://github.com/golang/crypto/compare/2c7772b...642fcc3
  * https://github.com/golang/sys/compare/a9b59b0...95e765b
* I scanned commit messages looking for changes to areas of interest, dependency changes, and/or breaking changes.  
* In my review, I did _not_ identify changes worth raising for further review.  Admittedly, _echo_ dependency changed _a lot_.  I'm mostly trusting that passing tests + no obvious commits for breaking behavior change impacting v4.1.17 mean no regression here.